types/wlr_layer_shell_v1.c

```diff
@@ -156,7 +156,18 @@ static void layer_surface_handle_set_keyboard_interactivity(
 	if (!surface) {
 		return;
 	}
-	surface->client_pending.keyboard_interactive = !!interactive;
+
+	if (wl_resource_get_version(resource) <= 3) {
```

Comment:
We can use `ZWLR_LAYER_SURFACE_V1_KEYBOARD_INTERACTIVITY_ON_DEMAND_SINCE_VERSION` instead (yeah, the constant name is a little bit long…).